pyrlang/node.py

```diff
@@ -369,64 +376,135 @@ def link_nowait(self, pid1, pid2, local_only=False):
         link_task = self.link(pid1, pid2, local_only)
         self._event_loop.create_task(link_task)
 
-    async def link(self, pid1, pid2, local_only=False):
-        """ Check each of processes pid1 and pid2 if they are local, mutually
-            link them. Assume remote process handles its own linking.
+    async def link(self, from_pid, to_pid, local_only=False):
+        """ Process a link event generated by from_pid, targeting to_pid.
 
-            :param pid1: First pid
-            :type pid1: term.pid.Pid
-            :param pid2: Second pid
-            :type pid2: term.pid.Pid
-            :param local_only: If set to True, linking to remote pids will send
-                LINK message over dist_proto protocol
+            :param from_pid: Source pid
+            :type from_pid: term.pid.Pid
+            :param to_pid: Target pid
+            :type to_pid: term.pid.Pid
+            :param local_only: Deprecated, ignored.
         """
-        if pid1.is_local_to(self):
-            if pid1 in self.processes_:
-                self.processes_[pid1].add_link(pid2)
-            else:
-                # not exists
-                await self._send_exit_signal(pid2, pid1, Atom("noproc"))
+        if from_pid.is_local_to(self) and from_pid not in self.processes_:
+            LOG.info("local from_pid %s not found during link", from_pid)
+            return
+        if to_pid.is_local_to(self) and to_pid not in self.processes_:
+            await self._send_exit_signal(to_pid, from_pid, Atom("noproc"))
+            return
 
-        elif not local_only:
-            link_m = ('link', pid2, pid1)
-            await self.dist_command(receiver_node=pid1.node_name_,
-                                    message=link_m)
+        from_process = self.processes_.get(from_pid, None)
```

Comment:
Here you do get, but below you just assume you got a process object, perhaps better to use `from_process = self.processes_[from_pid]` and get a key error here rather than AttributeError lower